StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1617CodeFixProvider.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace StyleCop.Analyzers.DocumentationRules
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1617VoidReturnValueMustNotBeDocumented"/>.
+    /// </summary>
+    /// <remarks>
+    /// <para>To fix a violation of this rule, remove the &lt;returns&gt; tag from the element.</para>
+    /// </remarks>
+    [ExportCodeFixProvider(nameof(SA1617CodeFixProvider), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1617CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+          ImmutableArray.Create(SA1617VoidReturnValueMustNotBeDocumented.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                if (!diagnostic.Id.Equals(SA1617VoidReturnValueMustNotBeDocumented.DiagnosticId))
+                {
+                    continue;
+                }
+
```

Comment:
I would prefer
```C#
foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
```
instead of the lines above. Its more generic and less clutter IMHO.